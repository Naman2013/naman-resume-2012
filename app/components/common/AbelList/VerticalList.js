import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './VerticalList.style';

const VerticalList = ({ list, iconList }) => (
  <Fragment>
    <ul>
      {list.map((entry, i) => (
        <li key={`vertical-list-${entry}`}>
          {iconList[i] && <img className="bullet-image" src={iconList[i]} />}{' '}
          {entry}
        </li>
      ))}
    </ul>
    <style jsx>{style}</style>
  </Fragment>
);

VerticalList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string),
  iconList: PropTypes.arrayOf(PropTypes.string),
};

VerticalList.defaultProps = {
  list: [],
  iconList: [],
};

export default VerticalList;
