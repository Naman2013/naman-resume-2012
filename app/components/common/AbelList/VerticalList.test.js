import React from 'react';
import { shallow } from 'enzyme';
import VerticalList from './VerticalList';

describe('VerticalList', () => {
  const sampleList = ['foo', 'bar', 'baz'];
  const shallowWrapper = shallow(<VerticalList list={sampleList} />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
