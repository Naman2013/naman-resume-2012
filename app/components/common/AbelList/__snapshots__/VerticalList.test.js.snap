// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalList should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VerticalList
    list={
      Array [
        "foo",
        "bar",
        "baz",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ul
          className="jsx-2156235474"
        >
          <li
            className="jsx-2156235474"
          >
            foo
          </li>
          <li
            className="jsx-2156235474"
          >
            bar
          </li>
          <li
            className="jsx-2156235474"
          >
            baz
          </li>
        </ul>,
        <JSXStyle
          css={
            Array [
              "ul.jsx-2156235474{margin:0;padding:0;list-style-type:none;}",
              "li.jsx-2156235474{text-transform:uppercase;font-size:10px;font-weight:bold;-webkit-letter-spacing:0.4px;-moz-letter-spacing:0.4px;-ms-letter-spacing:0.4px;letter-spacing:0.4px;padding:20px 0;border-bottom:1px solid #eaf0f9;font-family:brandon-grotesque, sans-serif;color:#415671;}",
            ]
          }
          styleId="2156235474"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="jsx-2156235474"
            >
              foo
            </li>,
            <li
              className="jsx-2156235474"
            >
              bar
            </li>,
            <li
              className="jsx-2156235474"
            >
              baz
            </li>,
          ],
          "className": "jsx-2156235474",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "vertical-list-foo",
            "nodeType": "host",
            "props": Object {
              "children": "foo",
              "className": "jsx-2156235474",
            },
            "ref": null,
            "rendered": "foo",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "vertical-list-bar",
            "nodeType": "host",
            "props": Object {
              "children": "bar",
              "className": "jsx-2156235474",
            },
            "ref": null,
            "rendered": "bar",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "vertical-list-baz",
            "nodeType": "host",
            "props": Object {
              "children": "baz",
              "className": "jsx-2156235474",
            },
            "ref": null,
            "rendered": "baz",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            "ul.jsx-2156235474{margin:0;padding:0;list-style-type:none;}",
            "li.jsx-2156235474{text-transform:uppercase;font-size:10px;font-weight:bold;-webkit-letter-spacing:0.4px;-moz-letter-spacing:0.4px;-ms-letter-spacing:0.4px;letter-spacing:0.4px;padding:20px 0;border-bottom:1px solid #eaf0f9;font-family:brandon-grotesque, sans-serif;color:#415671;}",
          ],
          "styleId": "2156235474",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ul
            className="jsx-2156235474"
          >
            <li
              className="jsx-2156235474"
            >
              foo
            </li>
            <li
              className="jsx-2156235474"
            >
              bar
            </li>
            <li
              className="jsx-2156235474"
            >
              baz
            </li>
          </ul>,
          <JSXStyle
            css={
              Array [
                "ul.jsx-2156235474{margin:0;padding:0;list-style-type:none;}",
                "li.jsx-2156235474{text-transform:uppercase;font-size:10px;font-weight:bold;-webkit-letter-spacing:0.4px;-moz-letter-spacing:0.4px;-ms-letter-spacing:0.4px;letter-spacing:0.4px;padding:20px 0;border-bottom:1px solid #eaf0f9;font-family:brandon-grotesque, sans-serif;color:#415671;}",
              ]
            }
            styleId="2156235474"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="jsx-2156235474"
              >
                foo
              </li>,
              <li
                className="jsx-2156235474"
              >
                bar
              </li>,
              <li
                className="jsx-2156235474"
              >
                baz
              </li>,
            ],
            "className": "jsx-2156235474",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "vertical-list-foo",
              "nodeType": "host",
              "props": Object {
                "children": "foo",
                "className": "jsx-2156235474",
              },
              "ref": null,
              "rendered": "foo",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "vertical-list-bar",
              "nodeType": "host",
              "props": Object {
                "children": "bar",
                "className": "jsx-2156235474",
              },
              "ref": null,
              "rendered": "bar",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "vertical-list-baz",
              "nodeType": "host",
              "props": Object {
                "children": "baz",
                "className": "jsx-2156235474",
              },
              "ref": null,
              "rendered": "baz",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              "ul.jsx-2156235474{margin:0;padding:0;list-style-type:none;}",
              "li.jsx-2156235474{text-transform:uppercase;font-size:10px;font-weight:bold;-webkit-letter-spacing:0.4px;-moz-letter-spacing:0.4px;-ms-letter-spacing:0.4px;letter-spacing:0.4px;padding:20px 0;border-bottom:1px solid #eaf0f9;font-family:brandon-grotesque, sans-serif;color:#415671;}",
            ],
            "styleId": "2156235474",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
